### PR TITLE
Condition muted should be saved with other persitent conditions

### LINKED ISSUE #2044: Reseting the mute's condition at death
### Before creating an issue, please ensure:
- [x] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [x] This issue is reproducible without changes to the code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. Mute a player
2. Kill the player

### Expected behaviour
The player should keep muted.

### Actual behaviour
When the player dies it reset the condition and the player can talk in the Channel which he was muted.

### Environment
The test was maded in Windows 7 - 64 bits.

### PR DESCRIPTION
**Pull request by thexamx**
Fix #2044

### CODE REVIEW

File: src/iologindata.cpp [hunk 1]
```diff
@@ -661,7 +661,7 @@ bool IOLoginData::savePlayer(Player* player)
 	//serialize conditions
 	PropWriteStream propWriteStream;
 	for (Condition* condition : player->conditions) {
-		if (condition->isPersistent()) {
+		if (condition->isPersistent() || condition->getType() == CONDITION_MUTED) {
```
Comment (DanielChabrowski) [inline src/iologindata.cpp]: Wouldn't the isPersistent function be a perfect place for that condition type check?

Comment (thexamx) [reply src/iologindata.cpp]: Do you mean
```
if (!(id == CONDITIONID_DEFAULT || id == CONDITIONID_COMBAT || getType() == CONDITION_MUTED)) {
	return false;
}
```
?

Comment (DanielChabrowski) [reply src/iologindata.cpp]: Yep.

File: src/condition.cpp [hunk 2]
```diff
@@ -266,7 +266,7 @@ bool Condition::isPersistent() const
 		return false;
 	}
 
-	if (!(id == CONDITIONID_DEFAULT || id == CONDITIONID_COMBAT)) {
+	if (!(id == CONDITIONID_DEFAULT || id == CONDITIONID_COMBAT || condition->getType() == CONDITION_MUTED)) {
```
Comment (DanielChabrowski) [inline src/condition.cpp]: `condition->getType()`
to
`conditionType`

Comment (thexamx) [reply src/condition.cpp]: I'm sorry, I did fast change and didnt notice.

**Comment by djarek** [comment]
Thanks!

**Pull request merged**